core/register_core_types.cpp

```diff
@@ -228,8 +228,8 @@ void register_core_types() {
 	GDREGISTER_CLASS(ResourceFormatLoader);
 	GDREGISTER_CLASS(ResourceFormatSaver);
 
-	GDREGISTER_CLASS(core_bind::File);
-	GDREGISTER_CLASS(core_bind::Directory);
+	GDREGISTER_ABSTRACT_CLASS(FileAccess);
+	GDREGISTER_ABSTRACT_CLASS(DirAccess);
```

Comment:
Shouldn't those be virtual instead of abstract?

They can be instantiated so (pure) abstract doesn't seem correct to me.

Reply:
I don't know what's the difference. Does virtual also prevent manual instantiation?
I checked how we use abstract and TreeItem is abstract, but you can instantiate it too.

Reply:
Ah so we want to forbid `FileAccess.new()` to force using a create method?

Then abstract might work indeed, but it feels to me like it's exploiting the wrong concept to expose a class which is not abstract nor virtual but just means to be non-instantiate directly.

Reply:
Yes, abstract is good in this case, if we want to provide something towards the future, in this case it will work out better as FileAccessExtension